internal/controller/promotions/promotions.go

```diff
@@ -176,6 +176,17 @@ func SetupReconcilerWithManager(
 		return fmt.Errorf("unable to watch Promotions: %w", err)
 	}
 
+	// Watch Stages that acknowledge their next Promotion and enqueue it
+	if err = c.Watch(
+		source.Kind(
+			kargoMgr.GetCache(),
+			&kargoapi.Stage{},
+			&PromotionAcknowledgedByStageHandler[*kargoapi.Stage]{},
+		),
+	); err != nil {
+		return fmt.Errorf("unable to watch Stages: %w", err)
+	}
+
```

Comment:
Nit: If this were moved up to somewhere around line 140, I feel like it would be a little easier to understand where this fits in with everything else.